src/test/java/tdl/runner/Steps.java

```diff
@@ -0,0 +1,225 @@
+        wiremockProcess.adjustStubMappingResponse(endpoint, returnValue);
+    }
+
+    // When
+
+    @When("user starts client")
+    public void userStartsChallenge() throws UnirestException {
+        printStream = new PrintStream(outContent);
+        wiremockProcess.configureServer();
+        String journeyId = "dGRsLXRlc3QtY25vZGVqczAxfFNVTSxITE8sQ0hLfFE=";
+        String username = "tdl-test-cnodejs01";
+        userCommandLineArgs = new String[]{};
+
+        ClientAction clientAction = new ClientAction() {
+            @Override
+            public void afterResponse(RemoteBroker remoteBroker, Request request, Response response) throws BrokerCommunicationException {
+
+            }
+
+            @Override
+            public Optional<Request> getNextRequest(RemoteBroker t) throws BrokerCommunicationException {
+                return Optional.empty();
+            }
+
+            @Override
+            public String getAuditText() {
+                return null;
+            }
+        };
+        UserImplementation userImplementation = new UserImplementation() {
+            @Override
+            public Object process(String... params) {
+                return null;
+            }
+        };
+        UserImplementation sum = new UserImplementation() {
+            @Override
+            public Object process(String... params) {
+                sumHit = true;
+                return null;
+            }
+        };
+        UserImplementation hello = new UserImplementation() {
+            @Override
+            public Object process(String... params) {
+                helloHit = true;
+                return null;
+            }
+        };
+        UserImplementation fizzBuzz = new UserImplementation() {
+            @Override
+            public Object process(String... params) {
+                fizzBuzzHit = true;
+                return null;
+            }
+        };
+        UserImplementation checkout = new UserImplementation() {
+            @Override
+            public Object process(String... params) {
+                checkoutHit = true;
+                return null;
+            }
+        };
+        Consumer<String> notifyRecordSystemCallback = new Consumer<String>() {
+            @Override
+            public void accept(String s) {
+                recordingSystemCallbackText = s;
+            }
+        };
+
+        // start in a background thread, check if thread finished for exit.
+
+        ClientRunner clientRunner = ClientRunner.forUsername(username)
+                .withServerHostname(hostname)
+                .withPort(port)
+                .withJourneyId(journeyId)
+                .withColours(true)
+                .withDeployCallback(() -> deployCallbackHit = true)
+                .withDeployAction(clientAction)
+                .withRecordingSystemOk(true)
+                .withBufferedReader(bufferedReader)
```

Comment:
I would not couple these names with the Java implementations. We need to be able to replicate them across languages.
I would introduce some interfaces: InputReader and OutputWriter